src/util/string.cpp

```diff
@@ -1065,3 +1065,30 @@ std::optional<v3f> str_to_v3f(std::string_view str)
 
 	return value;
 }
+
+std::string my_double_to_string(double number)
+{
+	if (std::isfinite(number)) {
+		char buf[64];
+		snprintf(buf, sizeof(buf), "%.17g", number);
+		return buf;
+	}
+	if (number < 0)
+		return "-inf";
+	if (number > 0)
+		return "inf";
+	return "nan";
+}
+
+std::optional<double> my_string_to_double(const std::string &s)
+{
+	if (s.empty())
+		return std::nullopt;
+	char *end = nullptr;
+	errno = 0;
+	// Note: this also supports hexadecimal notation like "0x1.0p+1"
+	double number = std::strtod(s.data(), &end);
```

Comment:
hopefully makes no practical difference
```suggestion
	double number = std::strtod(s.c_str(), &end);
```

Reply:
yeah as of C++ 11 it shouldn't. but best to use `c_str`, relieves the reader of any worries.

Reply:
looks like you unfixed this again? the force-pushing makes it kinda hard to keep track of the changes (same below)

Reply:
you're right. what must've happened is that i resolved these on the remote by applying sfan's suggestions, but hadn't pulled, so the force-push must've overwritten the commit. i'll fix them in a cleanup pr soon.

Reply:
there's also [`--force-with-lease`](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force-with-leaseltrefnamegtltexpectgt) :)